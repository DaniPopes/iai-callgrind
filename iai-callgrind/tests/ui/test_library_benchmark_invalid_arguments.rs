use iai_callgrind::library_benchmark;

#[library_benchmark(wrong = LibraryBenchmarkConfig::default())]
fn bench1() {}

#[library_benchmark(config = "wrong")]
fn bench2() {}

#[library_benchmark(config = )]
fn bench3() {}

fn main() {}
