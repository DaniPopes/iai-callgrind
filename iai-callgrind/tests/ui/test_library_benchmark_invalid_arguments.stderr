error: Only the config argument is allowed

         = help: #[library_benchmark(config = ....)]

 --> tests/ui/test_library_benchmark_invalid_arguments.rs:3:21
  |
3 | #[library_benchmark(wrong = LibraryBenchmarkConfig::default())]
  |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: unexpected end of input, expected an expression
 --> tests/ui/test_library_benchmark_invalid_arguments.rs:9:1
  |
9 | #[library_benchmark(config = )]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `library_benchmark` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `InternalLibraryBenchmarkConfig: From<&str>` is not satisfied
 --> tests/ui/test_library_benchmark_invalid_arguments.rs:6:1
  |
6 | #[library_benchmark(config = "wrong")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<&str>` is not implemented for `InternalLibraryBenchmarkConfig`
  |
  = help: the following implementations were found:
            <InternalLibraryBenchmarkConfig as From<&LibraryBenchmarkConfig>>
            <InternalLibraryBenchmarkConfig as From<&mut LibraryBenchmarkConfig>>
            <InternalLibraryBenchmarkConfig as From<LibraryBenchmarkConfig>>
  = note: required because of the requirements on the impl of `Into<InternalLibraryBenchmarkConfig>` for `&str`
  = note: this error originates in the attribute macro `library_benchmark` (in Nightly builds, run with -Z macro-backtrace for more info)
